lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```
@@ -174,6 +174,21 @@ public void Close(byte[] message)
                 _currentStatus.SetException(TaskClientCodeException.Create(
                     TaskId, ContextId, "Error during Close().", e));
             }
+            finally
+            {
+                try
+                {
+                    if (_userTask != null)
+                    {
+                        _userTask.Dispose();
+                    }
+                }
+                catch (Exception e)
+                {
+                    Utilities.Diagnostics.Exceptions.Caught(
```

This swallows the exception. Why is this a good idea? Also, if it is, we should document this behavior in the {{ITask}} interface documentation.